lib/magic-byte.js

```diff
@@ -45,24 +44,20 @@ magicByte.toMessageBuffer = function (val, type, schemaId, optLength) {
  *   @param {number} schemaId The schema id.
  *   @param {Object} value The decoded avro value.
  */
-magicByte.fromMessageBuffer = function (type, encodedMessage, sr) {
-  if (encodedMessage[0] !== MAGIC_BYTE) {
-    throw new TypeError('Message not serialized with magic byte');
+magicByte.fromMessageBuffer = function (encodedMessage, sr) {
+  if (!encodedMessage || encodedMessage[0] !== MAGIC_BYTE) {
+    return null;
```

Comment:
why are you returning null instead of throw TypeError?

Reply:
That because it's simpler to me to just check for null instead of doing a try catch block. But it doesn't matter that much to me and it's up to you preferred code style. If you would like I can convert it back to try-catch.